### PR TITLE
[Doc] Add Returns to Files

### PR DESCRIPTION
**Pull request by EsoragotoSpirit**
Signed-off-by: EsoragotoSpirit <wanglichen@starrocks.com>

## What I'm doing:

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [x] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [x] 3.2
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: docs/en/sql-reference/sql-functions/table-functions/files.md [hunk 1]
```diff
@@ -259,6 +259,107 @@ unload_data_param::=
...
 
+## Return
+
+When used with SELECT, FILES() returns the data in the file as a table.
+
+- When querying CSV files, you can use `$1`, `$2` ... to represent each column in the SELECT statement.
+
+  ```SQL
+  SELECT $1, $2 FROM FILES(
+    "path" = "s3://inserttest/csv/file1.csv",
```
Comment (jaogoy) [inline docs/en/sql-reference/sql-functions/table-functions/files.md:270]: 4 spaces indent

File: docs/en/sql-reference/sql-functions/table-functions/files.md [hunk 2]
```diff
@@ -259,6 +259,107 @@ unload_data_param::=
...
+    "format" = "csv",
+    "csv.column_separator"=",",
+    "csv.row_delimiter"="\r\n",
```
Comment (jaogoy) [inline docs/en/sql-reference/sql-functions/table-functions/files.md:273]: Use `\n`, most of the data are created on linux system, only `\n` exists.

File: docs/en/sql-reference/sql-functions/table-functions/files.md [hunk 3]
```diff
@@ -259,6 +259,107 @@ unload_data_param::=
...
+    "csv.enclose"='"',
+    "csv.skip_header"="1",
+    "aws.s3.access_key" = "XXXXXXXXXX",
```
Comment (jaogoy) [inline docs/en/sql-reference/sql-functions/table-functions/files.md:276]: We can use unified key/secret masks, it's much clear.

File: docs/en/sql-reference/sql-functions/table-functions/files.md [hunk 4]
```diff
@@ -259,6 +259,107 @@ unload_data_param::=
...
+  |    5 | 0.37766 |
+  |    6 | 0.34413 |
+  |    7 | 0.40055 |
+  |    8 | 0.42437 |
+  |    9 | 0.67935 |
+  |   10 | 0.22783 |
+  +------+---------+
+  10 rows in set (0.38 sec)
+
+  SELECT * FROM FILES(
```
Comment (jaogoy) [inline docs/en/sql-reference/sql-functions/table-functions/files.md:296]: Put this example ahead.

File: docs/en/sql-reference/sql-functions/table-functions/files.md [hunk 5]
```diff
@@ -259,6 +259,107 @@ unload_data_param::=
...
+  |  105 | 2018-05-15T18:30:00 |
+  |  106 | 2018-05-15T18:30:00 |
+  |  107 | 2018-05-15T18:30:00 |
+  |  108 | 2018-05-15T18:30:00 |
+  |  109 | 2018-05-15T18:30:00 |
+  |  110 | 2018-05-15T18:30:00 |
+  +------+---------------------+
+  10 rows in set (0.55 sec)
+
+  SELECT * FROM FILES(
```
Comment (jaogoy) [inline docs/en/sql-reference/sql-functions/table-functions/files.md:346]: Put this example ahead

File: docs/en/sql-reference/sql-functions/table-functions/files.md [hunk 6]
```diff
@@ -259,6 +259,107 @@ unload_data_param::=
 | single           | No           | Whether to unload the data into a single file. Valid values:<ul><li>`true`: The data is stored in a single data file.</li><li>`false` (Default): The data is stored in multiple files if the amount of data unloaded exceeds 512 MB.</li></ul>                  |
 | target_max_file_size | No           | The best-effort maximum size of each file in the batch to be unloaded. Unit: Bytes. Default value: 1073741824 (1 GB). When the size of data to be unloaded exceeds this value, the data will be divided into multiple files, and the size of each file will not significantly exceed this value. Introduced in v3.2.7. |
 
+## Return
+
+When used with SELECT, FILES() returns the data in the file as a table.
+
+- When querying CSV files, you can use `$1`, `$2` ... to represent each column in the SELECT statement.
```
Comment (jaogoy) [inline docs/en/sql-reference/sql-functions/table-functions/files.md:266]: or specify `*` to obtain data from all columns.

**Review by jaogoy — APPROVED** [review]

**Review by wangsimo0 — APPROVED** [review]

**Pull request merged**